src/browser/provider/built-in/dedicated/chrome/cdp-client/index.ts

```diff
@@ -168,6 +178,36 @@ export class BrowserClient {
         this._runtimeInfo.emulatedDevicePixelRatio = this._config.scaleFactor || this._runtimeInfo.originalDevicePixelRatio;
     }
 
+    private async _addScriptToEvaluateOnNewDocument (client: remoteChrome.ProtocolApi): Promise<void> {
+        await client.Page.addScriptToEvaluateOnNewDocument({
+            source: read('../../../../../../../lib/client/proxyless/index.js') as string
+        });
+    }
+
+    private _setupFramesWatching (client: remoteChrome.ProtocolApi): void {
+        // @ts-ignore
+        client.Runtime.executionContextCreated((event: Protocol.Runtime.ExecutionContextCreatedEvent) => {
+            if (!event.context.auxData?.frameId)
+                return;
+
+            this._frameExecutionContexts.set(event.context.auxData.frameId, event.context.id);
+        });
+
+        // @ts-ignore
+        client.Runtime.executionContextDestroyed((event: Protocol.Runtime.ExecutionContextDestroyedEvent) => {
+            for (const [frameId, executionContextId] of this._frameExecutionContexts.entries()) {
+                if (executionContextId === event.executionContextId)
+                    this._frameExecutionContexts.delete(frameId);
+            }
+        });
+
+        // @ts-ignore
+        client.Runtime.executionContextsCleared(() => {
```

Comment:
Could you please try to update `chrome-remote-interface` to `0.30.0`? This version seems to work with `client.Runtime.on()` syntax, supported by type defs.